### PR TITLE
Add reset password process

### PR DESCRIPTION
**Pull request by shyim**

### CODE REVIEW

File: src/Organization/System/Organization/Aggregate/OrganizationResetPasswordToken/OrganizationResetPasswordTokenDefinition.php [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+    protected function defineFields(): FieldCollection
+    {
+        return new FieldCollection([
+            (new IdField('id', 'id'))->addFlags(new PrimaryKey(), new Required()),
+
+            (new StringField('token', 'token'))->addFlags(new Required()),
+
+            (new FkField('organization_id', 'organizationId', OrganizationDefinition::class))->addFlags(new Required()),
+            (new ManyToOneAssociationField('organization', 'organization_id', OrganizationDefinition::class))
```
Comment (PaddyS) [inline src/Organization/System/Organization/Aggregate/OrganizationResetPasswordToken/OrganizationResetPasswordTokenDefinition.php:42]: Why `ManyToOne`?

Comment (PaddyS) [inline src/Organization/System/Organization/Aggregate/OrganizationResetPasswordToken/OrganizationResetPasswordTokenDefinition.php:42]: Why `ManyToOne`?

File: src/Organization/System/Organization/Api/ResetPasswordController.php [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+        if (!$dataBag->has('email')) {
+            throw new \InvalidArgumentException('Missing email request argument');
+        }
+
+        $criteria = new Criteria();
+        $criteria->addFilter(new EqualsFilter('email', $dataBag->get('email')));
+        $criteria->addAssociation('salesChannel.language.locale');
+
+        /** @var OrganizationEntity|null $organization */
+        $organization = $this->organizationRepository->search($criteria, Context::createDefaultContext())->first();
```
Comment (PaddyS) [inline src/Organization/System/Organization/Api/ResetPasswordController.php:90]: Can you check that email is unique?

File: src/Portal/Resources/snippets/de_DE/translation.json [hunk 3]
```diff
@@ -87,6 +87,12 @@
             "email": "E-Mail",
             "password": "Passwort",
             "loginButton": "Login"
+        },
+        "organization_reset_password": {
+            "subject": "Passwort Vergessen Anfrage bei Downtown",
```
Comment (PaddyS) [inline src/Portal/Resources/snippets/de_DE/translation.json:92]: Lowercase "vergessen"

File: src/Portal/Resources/snippets/de_DE/translation.json [hunk 4]
```diff
@@ -87,6 +87,12 @@
...
+            "hello": "Hallo %name%,",
+            "text": "im Portal wurde eine Anfrage gestellt, um Ihr Passwort zurückzusetzen.<br>\nBitte bestätigen Sie den untenstehenden Link, um ein neues Passwort zu definieren.",
```
Comment (PaddyS) [inline src/Portal/Resources/snippets/de_DE/translation.json:94]: Is "Portal" the right name?

File: src/Organization/System/Organization/Aggregate/OrganizationResetPasswordToken/OrganizationResetPasswordTokenDefinition.php (hunk 1 above)
Comment (shyim) [reply src/Organization/System/Organization/Aggregate/OrganizationResetPasswordToken/OrganizationResetPasswordTokenDefinition.php:42]: es können mehrere tokens existieren habs von merchant kopiert xD

File: src/Organization/System/Organization/Api/ResetPasswordController.php (hunk 2 above)
Comment (shyim) [reply src/Organization/System/Organization/Api/ResetPasswordController.php:90]: That field is unique in the database

File: src/Portal/Resources/snippets/de_DE/translation.json (hunk 4 above)
Comment (shyim) [reply src/Portal/Resources/snippets/de_DE/translation.json:94]: I dont know

**Pull request merged**